src/Application/Features/Participants/EventHandlers/ConsentCreated.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Cfo.Cats.Domain.Events;
+
+namespace Cfo.Cats.Application.Features.Participants.EventHandlers;
+
+public class ConsentCreated(IUnitOfWork unitOfWork):INotificationHandler<ConsentCreatedDomainEvent>
+{
+    public async Task Handle(ConsentCreatedDomainEvent notification, CancellationToken cancellationToken)
+    {
+        var participant = await unitOfWork.DbContext.Participants
+            .Where(p => p.Id == notification.ParticipantId)
+            .FirstOrDefaultAsync(cancellationToken);
+
+        if (participant is not null && participant.Consents.Count == 1)
```

Comment:
```suggestion
        if (participant is { Consents.Count: 1 })
```
You could use pattern matching here